sklearn/sklearn-template/template/trainer/task.py

```diff
@@ -65,13 +65,13 @@ def _train_and_evaluate(estimator, dataset, output_dir):
   trial_id = str(hpt.trial_id)
 
   # Write model and eval metrics to `output_dir`
-  model_file_name = '{}_{}{}'.format(metadata.MODEL_FILE_NAME_PREFIX,
-                                     timestamp, metadata.MODEL_FILE_NAME_SUFFIX)
-  model_output_path = os.path.join(output_dir, trial_id, model_file_name)
-  metric_file_name = '{}_{}{}'.format(
-      metadata.METRIC_FILE_NAME_PREFIX,
-      timestamp, metadata.MODEL_FILE_NAME_SUFFIX)
-  metric_output_path = os.path.join(output_dir, trial_id, metric_file_name)
+  trial_folder = os.path.join(output_dir, '{}_{}'.format(trial_id, timestamp))
```

Comment:
I'm not sure that we would need to keep track of trials. I think CMLE hyperparameter tuning already takes care of it.

Reply:
We need to track it in order to make sure the trained model is exported to the folder accordingly. Then, after the hyperparameter tuning is done, we can easily recover the best model. Tensorflow has Exporter that can help to do it.

Reply:
I understand your intent. From my experience with running hptuning jobs I noticed that CMLE would create subfolders in `MODEL_DIR` with indexes as names for the different trials. Each folder would be like its own MODEL_DIR with the model files, e.g.
```
gs://bucket/model_dir/0
gs://bucket/model_dir/1
...
gs://bucket/model_dir/<n>
```

This goes for both TF and sklearn CMLE jobs. So I don't think we would need to track the trials and put it into folders ourselves.

Reply:
It is because there is an `exporter = tf.estimator.FinalExporter()` defined, which will help to export the model properly. But for our cases, we are in control of the exporting process.